Sources/SwiftDocC/Infrastructure/DocumentationCurator.swift

```diff
@@ -242,19 +257,48 @@ struct DocumentationCurator {
                     let hasTechnologyRoot = isTechnologyRoot(nodeReference) || context.reachableRoots(from: nodeReference).contains(where: isTechnologyRoot)
 
                     if !hasTechnologyRoot {
-                        problems.append(Problem(diagnostic: Diagnostic(source: source(), severity: .warning, range: range(), identifier: "org.swift.docc.ModuleCuration", summary: "Linking to \((link.destination ?? "").singleQuoted) from a Topics group in \(nodeReference.absoluteString.singleQuoted) isn't allowed", explanation: "The former is a module, and modules only exist at the root"), possibleSolutions: []))
+                        problems.append(Problem(
+                            diagnostic: Diagnostic(
+                                source: source(), severity: .warning, range: range(), identifier: "org.swift.docc.ModuleCuration",
+                                summary: "Organizing the module \(childReference.lastPathComponent.singleQuoted) under \(describeForDiagnostic(nodeReference).singleQuoted) isn't allowed",
+                                explanation: "\(topicSectionBaseExplanation). Modules should be roots in the documentation hierarchy."),
+                            possibleSolutions: removeListItemSolutions))
                         continue
                     }
                 }
                 
                 // Verify we are not creating a graph cyclic relationship.
                 guard childReference != nodeReference else {
-                    problems.append(Problem(diagnostic: Diagnostic(source: source(), severity: .warning, range: range(), identifier: "org.swift.docc.CyclicReference", summary: "A symbol can't link to itself from within its Topics group in \(nodeReference.absoluteString.singleQuoted)"), possibleSolutions: []))
+                    problems.append(Problem(
+                        diagnostic: Diagnostic(
+                            source: source(), severity: .warning, range: range(), identifier: "org.swift.docc.CyclicReference",
+                            summary: "Organizing \(describeForDiagnostic(childReference).singleQuoted) under itself forms a cyclic documentation hierarchy",
```

Comment:
Maybe simplify the message to: "...under itself forms a cycle"

Reply:
I updated this to say "form a cycle" in https://github.com/swiftlang/swift-docc/pull/984/commits/a2c52e0d4a12774d5d7c047efe16d2ac86c17441